index.js

```diff
@@ -1 +1,105 @@
 // implement your API here
+const express = require("express");
+const Db = require("./data/db");
+
+const server = express();
+
+server.use(express.json());
+
+server.get("/", (req, res) => {
+  res.send("Monday July 8 WebAPI Challenge");
+});
+
+server.get("/api/users", function(req, res) {
+  Db.find()
+    .then(users => {
+      res.status(200).json(users);
+    })
+    .catch(error => {
+      res
+        .status(500)
+        .json({ error: "The users information could not be retrieved." });
+    });
+});
+
+server.get("/api/users/:id", function(req, res) {
+  const { id } = req.params;
+  console.log(id);
+  Db.findById(id)
+    .then(user => {
+      res.status(200).json(user);
+    })
+    .catch(error => {
+      res
+        .status(404)
+        .json({ message: "The user with the specified ID does not exist." });
+    });
+});
+
+server.post("/api/users", (req, res) => {
+  const userInfo = req.body;
+  console.log(userInfo);
+
+  Db.insert(userInfo)
+    .then(user => {
+      if (user) {
+        res.status(204).send(user);
+      } else {
+        res
+          .status(400)
+          .json({ errorMessage: "Please provide name and bio for the user." });
+      }
+    })
+    .catch(error => {
+      res.status(500).json({
+        error: "There was an error while saving the user to the database"
+      });
+    });
+});
+
+server.delete("/api/users/:id", (req, res) => {
+  const { id } = req.params;
+
+  Db.remove(id)
+    .then(deleted => {
+      if (deleted === 1) {
+        res.status(204).end();
+      } else {
+        res
+          .status(404)
+          .json({ message: "The user with the specified ID does not exist." });
+      }
+    })
+    .catch(error => {
+      res.status(500).json({ error: "The user could not be removed" });
+    });
+});
+
+server.put("/api/users/:id", (req, res) => {
+  const id = req.params.id;
+  const changes = req.body;
+
+  Db.update(id, changes)
+    .then(updated => {
+      if (updated) {
+        res.status(200).json(updated);
```

Comment:
Here, on success, we are supposed to return the newly updated user document. However, Db.update returns a 0 or 1 depending on whether or not a document was updated. So how are we to find the newly updated object? We'll use the id we got from the params instead: 
```suggestion
        Db.findById(id).then(user => res.status(200).json(user));
```